pkg/handlers/scripts/install.sh

```diff
@@ -1,116 +1,352 @@
+	case $(uname -m) in
+		"amd64" | "x86_64")
+			ARCH="amd64"
+			;;
+		"arm64" | "aarch64")
+			ARCH="arm64"
+			;;
+		"arm")
+			ARCH="arm"
+			;;
+		"i386")
+			ARCH="386"
+			;;
+		*)
+			fail "unsupported architecture $(uname -m)"
+	esac
+	[[ ! -z "${ARCH+x}" ]] || fail "could not determine the architecture"
+
+	# Check for the current OS + arch combination in the available assets.
+	# NOTE: the case statements are built by the templating engine by ranging
+	#   over the set of assets and creating an OS_ARCH case that assigns that
+	#   asset's URL and file type.
+	case "${OS}_${ARCH}" in
+	  {{ range .Assets }}
+		{{ .OS }}_{{ .Arch }})
+			URL="{{ .URL }}"
+			FTYPE="{{ .Type }}"
+			;;
+		{{ end }}
+		*)
+			fail "No asset found for platform ${OS}-${ARCH}"
+			;;
+	esac
+	[[ ! -z "${URL+x}" || ! -z "${FTYPE+x}" ]] || fail "could not find a valid release URL for ${OS} ${ARCH}"
+
+	# Check for needed utilities.
+	command -v curl &> /dev/null || fail "curl not installed"
+	command -v find &> /dev/null || fail "find not installed"
+	command -v xargs &> /dev/null || fail "xargs not installed"
+	command -v sort &> /dev/null || fail "sort not installed"
+	command -v tail &> /dev/null || fail "tail not installed"
+	command -v cut &> /dev/null || fail "cut not installed"
+	command -v du &> /dev/null || fail "du not installed"
+
+	# Check that the assets can be extracted.
+	case "${FTYPE}" in
+		".gz")
+			command -v gzip &>/dev/null || fail "gzip is not installed"
+			;;
+		".tar.gz")
+			command -v tar &>/dev/null || fail "tar is not installed"
+			;;
+		".zip")
+			command -v unzip &>/dev/null || fail "zip is not installed"
+			;;
+		"")
+			;;
+		*)
+			fail "unsupported file type ${FTYPE}"
+	esac
+
+	# Check if the install directory needs to be prompted for and exists.
+	if [[ -z "${OUT_DIR:+x}" ]]; then
+		 prompt_install_dir
+	fi
+
+	if [[ ! -d "${OUT_DIR}" ]]; then
+		if [[ ! -z "${USE_SUDO+x}" ]]; then
+			sudo mkdir -p "${OUT_DIR}" &> /dev/null || true
+		else
+			mkdir -p "${OUT_DIR}" &> /dev/null || true
+		fi
 	fi
-	#search subtree largest file (bin)
+
+	if [[ ! -w "${OUT_DIR}" && -z "${USE_SUDO+x}" ]]; then
+		echo ""
+		echo "The installation directory ${OUT_DIR} is not writeable by this user, and installation has failed."
+		echo ""
+		echo "To fix this, do one of the following:"
+		echo "  * Set the environment variable REPL_INSTALL_PATH to a directory in the PATH that can"
+		echo "    be written to and re-run this script"
+		echo "  * Set the environment variable REPL_USE_SUDO to any value and re-run this script. Keep"
+		echo "    in mind this script will block waiting on sudo."
+		echo "  * Re-run this script with the -i or --install flag set to a directory in the PATH that"
+		echo "    can be written to. 'curl http://kots.io | bash -s --install /new/path'"
+		echo "  * Re-run this script with the -s or --sudo flag. Keep in mind that this script will"
+		echo "    block waiting on sudo. 'curl http://kots.io | bash -s --sudo'"
```

Comment:
Saw your note:
> I may be wrong, but I think the bash script cannot read command line arguments if it is piped into bash.

Should we modify or remove these instructions if that is the case?